### ISSUE TITLE
Inference phase crash with TF 1.4.1 and Keras 2.0.8

### ISSUE THREAD

**Issue by arthurPrvst** [matterport/Mask_RCNN#880]
I'm facing issues with using this repo with **TF 1.4.1 and Keras 2.0.8.**

In order to make the training phase works I have to change :
- `K.int_shape(x) to K.shape(x)` line 967 in `model.py
`-` loss = tf.nn.sparse_softmax_cross_entropy_with_logits(labels=target_class_ids, logits=pred_class_logits)` to 
```
new = tf.transpose(target_class_ids)
new = tf.reshape(new, [1, -1])
loss = tf.nn.sparse_softmax_cross_entropy_with_logits(labels=new, logits=pred_class_logits)

```
In order to make inference phase works I have to change : 
`indices = tf.stack([tf.range(probs.shape[0]), class_ids], axis=1)` to 
`indices = tf.stack([tf.range(config.POST_NMS_ROIS_INFERENCE), class_ids], axis=1)` line 716.

This modification in refine_detections_graph allow me to avoid this error : `"Cannot convert an unknow Dimension to a Tensor: %s" % d)`", but when I use `inspect_ablloon_model `notebook, with `mask_rcnn_ballon.h5 `weights, network doesn't make prediction.
Any idea why the network doesn't not predict bbox/segmentation ?
Is this related to modifications I did cause of my TF version ?

Thanks in advance,

**Comment by arthurPrvst** [comment]
Errors described above comes from TF version (1.4.1). **I updated my `TF to 1.8` (with` Cuda 8` and `Cudnn`), and errors disapeared.** Correction I did for 1.4.1 allow us to bypass errors and to run the project, but they don't give us a correct model.
